src/common/http/http.cpp

```diff
@@ -115,11 +115,24 @@ error::Error BreakDownUrl(const string &url, BrokenDownUrl &address) {
 		address.path = tmp.substr(split_index);
 	}
 
-	if (address.host.find("@") != string::npos) {
-		address = {};
-		return error::Error(
-			make_error_condition(errc::not_supported),
-			"URL Username and password is not supported");
+	auto auth_index = address.host.find("@");
```

Comment:
The Golang URL parser allows the username/password to contain `@` and instead uses the [last occurrence](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/net/url/url.go;l=587) of the character as the authority/host separator.
I know that `wget` and `curl` does not allow it, and will confuse the `host:port` section if you include additional `@` characters. In both cases it also works to URL escape `@` using `%40`.